base/deprecated.jl

```diff
@@ -148,8 +148,6 @@ end
 @deprecate inf{T<:AbstractFloat}(::Type{T})  convert(T,Inf)
 @deprecate nan{T<:AbstractFloat}(::Type{T})  convert(T,NaN)
 
-@deprecate_binding String AbstractString
```

Comment:
So we haven't had a release cycle where String was an error. By the policy I think we'd need to before giving it a new meaning.